docker-compose.prod.yml

```diff
@@ -0,0 +1,175 @@
+    environment:
+      - NODE_NAME=node1
+      - NODE_PORT=2428
+      - SWARM_PORT=2528
+    ports:
+      - "2428:2428"
+      - "2528:2528"
+    entrypoint: ["/bin/sh", "-c"]
+    command: >
+      "mkdir -p /calimero/data &&
+      if [ ! -d /calimero/data/config ]; then
+        merod --node-name node1 --home /calimero/data init --server-host 0.0.0.0 --server-port 2428 --swarm-port 2528
+      fi &&
+      merod --node-name node1 --home /calimero/data run"
+    depends_on:
+      init_volume:
+        condition: service_completed_successfully
+      auth:
+        condition: service_healthy
+    networks:
+      - internal
+      - web
+    labels:
+      - "traefik.enable=true"
+
+      # JSON-RPC route
+      - "traefik.http.routers.node-api.rule=PathPrefix(`/jsonrpc`) || PathPrefix(`/admin-api`)"
+      - "traefik.http.routers.node-api.entrypoints=web"
+      - "traefik.http.routers.node-api.service=node-core"
+      - "traefik.http.routers.node-api.middlewares=cors,auth-node"
+
+      # Admin dashboard route
+      - "traefik.http.routers.node-dashboard.rule=PathPrefix(`/admin-dashboard`)"
+      - "traefik.http.routers.node-dashboard.entrypoints=web"
+      - "traefik.http.routers.node-dashboard.service=node-core"
+      - "traefik.http.routers.node-dashboard.middlewares=cors"
+
+      # Forward Auth middleware for node service
+      - "traefik.http.middlewares.auth-node.forwardauth.address=http://auth:3001/auth/validate"
+      - "traefik.http.middlewares.auth-node.forwardauth.trustForwardHeader=true"
+      - "traefik.http.middlewares.auth-node.forwardauth.authResponseHeaders=X-Auth-User,X-Auth-Permissions"
+
+      # CORS middleware
+      - "traefik.http.middlewares.cors.headers.accesscontrolallowmethods=GET,OPTIONS,PUT,POST,DELETE"
+      - "traefik.http.middlewares.cors.headers.accesscontrolallowheaders=*"
+      - "traefik.http.middlewares.cors.headers.accesscontrolalloworiginlist=*"
+      - "traefik.http.middlewares.cors.headers.accesscontrolmaxage=100"
+      - "traefik.http.middlewares.cors.headers.addvaryheader=true"
+      - "traefik.http.middlewares.cors.headers.accesscontrolexposeheaders=X-Auth-Error"
+
+      # Define the service
+      - "traefik.http.services.node-core.loadbalancer.server.port=2428"
+
+  # Auth service
+  auth:
+    image: calimero/auth:dev
+    build:
+      context: .
+      dockerfile: Dockerfile.auth
+      target: dev  # Use the development stage
+      args:
+        - CALIMERO_AUTH_FRONTEND_FETCH=1
+    volumes:
+      - .:/usr/src/calimero:cached  # Mount the entire source code
+      - cargo-registry:/usr/local/cargo/registry  # Cache cargo registry
+      - cargo-git:/usr/local/cargo/git  # Cache git repositories
+      - target:/usr/src/calimero/target  # Cache build artifacts
+      - calimero_auth_data:/data/auth_db  # Persist RocksDB data
+    environment:
+      - RUST_LOG=debug
+      - CARGO_TARGET_DIR=/usr/src/calimero/target
+      - CALIMERO_AUTH_FRONTEND_FETCH=1
```

Comment:
these are build envs in runtime env position, and do we need either of them? you already set the build WORKDIR in the dockerfile